src/worker/network.rs

```diff
@@ -379,10 +394,19 @@ fn get_local_addr_port(
     let local_port = 49152 + (index % conns_per_addr as u64) as u16;
     debug!("addr {}, index {}", addr, index);
 
+    // conns_per_addr effectively groups connections together, one address per
+    // group with only port being different. addr_index represent current index
+    // inside the space of such groups.
     let mut addr_index = index / conns_per_addr as u64;
     let mut octets = addr.0;
 
-    info!("addr_index= {}", addr_index);
+    // The first resulting local address must not collide with the server
+    // addres, thus shift it by one right away.
+    if octets[3] != 255 {
```

Comment:
If the IPv4 addres is a /24, then 255 would be the broadcast IP and I don't think we want that, might want to cap at 254.

Reply:
Yeah, I need to test what can happen in that case. But it's probably not a big deal for now, and could be addressed later.